test/bitset.jl

```diff
@@ -66,9 +66,11 @@ end
 end
 
 @testset "issue #8570" begin
-    @test 400 > @allocated s = BitSet(typemax(Int32))
-    @test length(s) === 1
-    for b in s; b; end
+    let s
+        @test 400 > @allocated s = BitSet(typemax(Int32))
+        @test length(s) === 1
+        for b in s; b; end
```

Comment:
Is this testing that iteration doesn't crash? Maybe replace this by `@test only(s) == typemax(Int32)`, or `iterate(s)[1] == typemax(Int32)` ?

Reply:
In the original issue, this iteration ran forever, so I assume that this is testing for that. `only` should work better.